services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java

```diff
@@ -156,14 +170,20 @@ private static final class AttachHttpAttributesExecutionInterceptor implements E
         @Override
         public void afterMarshalling(Context.AfterMarshalling context,
                                      ExecutionAttributes executionAttributes) {
-            SdkHttpExecutionAttributes attributes =
+
+            SdkHttpExecutionAttributes.Builder attributes =
                 SdkHttpExecutionAttributes.builder()
                                           .put(OPERATION_NAME,
                                                executionAttributes.getAttribute(SdkExecutionAttribute.OPERATION_NAME))
-                                          .build();
+                    .put(HTTP_CHECKSUM, executionAttributes.getAttribute(SdkInternalExecutionAttribute.HTTP_CHECKSUM));
```

Comment:
I can't find where the `SdkInternalExecutionAttribute.HTTP_CHECKSUM` is set in the PR?

Reply:
It's set in the generated service code based on the service model.